modules/comment/comment.service.js

```diff
@@ -50,26 +50,26 @@ class CommentsService {
     const product = await Product.findById(id);
     const { userRates } = product;
     let { rateCount } = product;
-    const { rate } = userRates.find(rate => String(rate.user) === String(user._id)) || {};
+    const { rate } =      userRates.find(rate => String(rate.user) === String(user._id)) || {};
```

Comment:
I think it is better to have two different flows, where rate exists and where we need to create a rate object from scratch, it will be much more readable

Reply:
it is not my component